src/agent/state-inspector.ts

```diff
@@ -0,0 +1,533 @@
+    let scriptId: string = frame.location.scriptId;
+    if (this.scriptmapper_[scriptId] === undefined) {
+      return '';
+    }
+    if (this.scriptmapper_[scriptId].url === undefined) {
+      return '';
+    }
+    return this.scriptmapper_[scriptId].url;
+  }
+
+  resolveRelativePath_(frame: inspector.Debugger.CallFrame): string {
+    const fullPath = this.resolveFullPath_(frame);
+    return this.stripCurrentWorkingDirectory_(fullPath);
+  }
+
+  stripCurrentWorkingDirectory_(path: string): string {
+    // Strip 1 extra character to remove the slash.
+    // TODO: Handle the case where `this.config_.workingDirectory` is `null`.
+    return path.substr((this.config_.workingDirectory as string).length + 1);
+  }
+
+  isPathInCurrentWorkingDirectory_(path: string): boolean {
+    // return true;
+    // TODO: Handle the case where `this.config_.workingDirectory` is `null`.
+    return path.indexOf(this.config_.workingDirectory as string) === 0;
+  }
+
+  isPathInNodeModulesDirectory_(path: string): boolean {
+    return path.indexOf('node_modules') === 0;
+  }
+
+  resolveFrame_(frame: inspector.Debugger.CallFrame, underFrameCap: boolean):
+      apiTypes.StackFrame {
+    let args: Array<apiTypes.Variable> = [];
+    let locals: Array<any> = [];
+
+    if (!underFrameCap) {
+      args.push({
+        name: 'arguments_not_available',
+        varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+      });
+      locals.push({
+        name: 'locals_not_available',
+        varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+      });
+    } else {
+      args = [];
+      locals = this.resolveLocalsList_(frame);
+
+      if (isEmpty(locals)) {
+        locals = [];
+      }
+    }
+    return {
+      function: this.resolveFunctionName_(frame),
+      location: this.resolveLocation_(frame),
+      arguments: args,
+      locals: locals
+    };
+  }
+
+  resolveFunctionName_(frame: inspector.Debugger.CallFrame): string {
+    if (!frame) {
+      return '';
+    }
+    if (frame.functionName === '') {
+      return '(anonymous function)';
+    }
+    return frame.functionName;
+  }
+
+  resolveLocation_(frame: inspector.Debugger.CallFrame):
+      apiTypes.SourceLocation {
+    return {
+      path: this.resolveRelativePath_(frame),
+      line: frame.location.lineNumber
+    };
+  }
+
+  /**
+   * Iterates and returns variable information for all scopes (excluding global)
+   * in a given frame. FrameMirrors should return their scope object list with
+   * most deeply nested scope first so variables initially encountered will take
+   * precedence over subsequent instance with the same name - this is tracked in
+   * the usedNames map. The argument list given to this function may be
+   * manipulated if variables with a deeper scope occur which have the same
+   * name.
+   * @function resolveLocalsList_
+   * @memberof StateResolver
+   * @param {FrameMirror} frame - A instance of FrameMirror
```

Comment:
Change the JSDoc to reflect the new argument type.

Reply:
Done.